RemoveLinksToAttachments.py

```diff
@@ -22,69 +25,54 @@ def select_references(db):
                          " AND REFERENCED_COLUMN_NAME LIKE 'AttachmentID' and table_name != 'attachment'")
     return db_reference.fetchall()
 
-# writes the data passed in to a csv file
-def save_report(file_name,data):
-    write_report(file_name,["Attachment ID"],data)
-    print("Report saved as '%s.csv'" % file_name)
-
-# deletes any entry that references the attachmentID, then deletes attachmentID from the schema, handels any conflicts
+# deletes any entry that references the attachmentID, then deletes attachmentID from the schema, handles any conflicts
 def delete_attachments(db,attachments):
     db_delete_reference = db.cursor()
     db_delete_attachment = db.cursor()
-    conflicts = []
     for record in attachments:
-        try:
-            for table in select_references(db):
-                db_delete_reference.execute("DELETE FROM %s WHERE AttachmentID = %s" % (table[0], record[0]))
-            db_delete_attachment.execute("DELETE FROM attachment WHERE AttachmentID = %s" % record[0])
-            db.commit()
-        except:
-            conflicts += [record[0]]
-    if len(conflicts) == 0:
-        print(attachments)
-        print("%s attachments deleted" % len(attachments))
-        print("no conflicts")
-        return
-    print("%s attachments deleted" % (int(len(attachments))-int(len(conflicts))))
-    display = input("Show or save %s conflict(s)? [show/save/n]" % len(conflicts))
-    if display == "show":
-        for c in conflicts:
-            print(c)
-    elif display == "save":
-        save_report("AttachmentConflicts",conflicts)
-    elif display == "n":
-        pass
-    else:
-        print("Invalid command")
+        for table in select_references(db):
```

Comment:
why did you remove the exception handling here? How does the user know if one or more of the deletes fail?

Reply:
Script has been reformatted to include exception handling 
https://github.com/AAFC-BICoE/specify-scripts/blob/f1c1ce7bae43fde6c4cef3543d95ead5bdbd412d/RemoveLinksToAttachments.py#L29-L38